apps/worker/src/app/workflow/usecases/send-message/send-message-chat.usecase.ts

```diff
@@ -167,33 +177,50 @@ export class SendMessageChat extends SendMessageBase {
       filterData: {
         tenant: command.job.tenant,
       },
-    });
+    };
 
-    if (!integration) {
-      await this.executionLogRoute.execute(
-        ExecutionLogRouteCommand.create({
-          ...ExecutionLogRouteCommand.getDetailsFromJob(command.job),
-          detail: DetailEnum.SUBSCRIBER_NO_ACTIVE_INTEGRATION,
-          source: ExecutionDetailsSourceEnum.INTERNAL,
-          status: ExecutionDetailsStatusEnum.FAILED,
-          isTest: false,
-          isRetry: false,
-          raw: JSON.stringify({
-            reason: `Integration with integrationId: ${subscriberChannel?._integrationId} is either deleted or not active`,
-          }),
-        })
-      );
+    /**
+     * Current a workaround as chat providers for whatsapp is more similar to sms than to our chat implementation
```

Comment:
Would you consider inverting the logic for this workaround to make it easier to reason about? That is, build the command payload without an ID and add the provider is not WhatsApp.

Reply:
Alternatively you can redefine a `getIntegration` method on this class and encapsulate the workaround only in that method.